qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/VectorSearchIT.java

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.upgrades;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.equalTo;
+
+public class VectorSearchIT extends ParameterizedRollingUpgradeTestCase {
+    public VectorSearchIT(@Name("upgradedNodes") int upgradedNodes) {
+        super(upgradedNodes);
+    }
+
+    private static final String FLOAT_INDEX_NAME = "float_vector_index";
+    private static final String SCRIPT_VECTOR_INDEX_NAME = "script_vector_index";
+    private static final String BYTE_INDEX_NAME = "byte_vector_index";
+    private static final String QUANTIZED_INDEX_NAME = "quantized_vector_index";
+    private static final String FLOAT_VECTOR_SEARCH_VERSION = "8.4.0";
+    private static final String BYTE_VECTOR_SEARCH_VERSION = "8.6.0";
+    private static final String QUANTIZED_VECTOR_SEARCH_VERSION = "8.12.1";
+
+    public void testScriptVectorSearch() throws Exception {
+        assumeTrue(
+            "Float vector search is not supported on this version",
+            getOldClusterTestVersion().onOrAfter(FLOAT_VECTOR_SEARCH_VERSION)
+        );
+        if (isOldCluster()) {
```

Comment:
One silly question as I'm not really familiar with this new framework. Will this  be invoked during all steps in a strict order (old cluster / mixed / upgraded) so with this check here we ensure that the index creation  will only run once?